pandas/io/parsers.py

```diff
@@ -2280,18 +2262,11 @@ def __init__(self, f, **kwds):
         self.comment = kwds["comment"]
         self._comment_lines = []
 
-        self.handles = get_handle(
-            f,
-            "r",
-            encoding=self.encoding,
-            compression=self.compression,
-            memory_map=self.memory_map,
-        )
-
         # Set self.data to something that can read lines.
         if hasattr(self.handles.handle, "readline"):
             self._make_reader(self.handles.handle)
         else:
+            # read_excel: self.handles.handle is a list
             self.data = self.handles.handle
```

Comment:
if `self.handles.handle` is a list (or has no `readline`) `PythonParser` will treat the 'file handle'  as data